docs/onboarding_best_practices.rst

```diff
@@ -0,0 +1,165 @@
+
+
+
+
+Realtime specific considerations
+*********************************
+
+1. Time column
+###################
+Time column is a must for hybrid table
+
+2. Num partitions
+###################
+When creating the stream topic (typically for your output of Samza job),  keep the number of partitions to a minimum. If partitions are increased later, Pinot is able to automatically and seamlessly start consuming from them.
+
+3. Retention
+###################
+The retention of the stream topic should be set depending on the lag in your offline push. In hybrid tables, depending on the push frequency (set in table config), Pinot will start using the offline data 
+
+- 1 day after offline data becomes available, if push frequency daily 
+- 1 hour after offline data becomes available, if push frequency is hourly
+
+Generally a stream retention of 4-5 days is sufficient, as this is enough time for the offline flow to generate data. The retention of the realtime table should no be more than the stream retention.
+
+4. Pre aggregations
+###################
+Consider reducing the number of stream events per second, by aggregating in a samza job. Keep the aggregation window reasonable (5 minutes, 15 minutes, 1 hour depending on the events per second). If the aggregation window is large (say 6 hours), a burst of events will be sent to Pinot periodically after long gaps of silence. This kind of ingestion pattern is not ideal for Pinot consumption, and a steady stream of events is preferred. 
+
+5. Metrics aggregation in Pinot realtime
+#########################################################
+If aggregation is not possible at source (due to freshness concerns), pinot can aggregate after events are received, by enabling aggregateMetrics: true in table config. Note that Pinot supports only “sum” aggregations in realtime.
+
+6. Aggregation buckets
+#####################################
+Keep the aggregation bucket size the same across realtime and offline. If aggregation buckets are not consistent across realtime and offline, there can be undesirable results for data which spans the boundary. For example, if in realtime you have MILLISECONDS time column rounded off to the nearest hour via the samza job, follow the same model in offline flow, and round off time column to the nearest hour in the pre aggregations. This means, the samza job will produce a new timestamp every hour, whereas the offline segment for the day will have 24 unique timestamps. 
+
+
+
+
+Querying Pinot
+^^^^^^^^^^^^^^
+`PQL doc <https://pinot.readthedocs.io/en/latest/pql_examples.html>`_
+
+1. Apply **time filters** to your queries where applicable, to avoid fetching too much data
+
+2. Apply **reasonable LIMIT and TOP** to your query results, based on what application needs
```

Comment:
Not sure how an application writer can use this.  Would it be ok to word it like "Fetching a lot of data from Pinot is sub-optimal, so take care when specifying values for TOP and LIMIT clauses"